python/pyarrow/_flight.pyx

```diff
@@ -736,16 +742,48 @@ cdef class FlightEndpoint(_Weakrefable):
                     CLocation.Parse(tobytes(location)).Value(&c_location))
             self.endpoint.locations.push_back(c_location)
 
+        if expiration_time is not None:
+            self.endpoint.expiration_time = TimePoint_to_system_time(TimePoint_from_ns(
+                expiration_time.cast(timestamp("ns")).value))
+
+        self.endpoint.app_metadata = tobytes(app_metadata)
```

Comment:
sorry, does this handle None appropriately? (The docstring states 'optional')

Reply:
Optional in the sense that there is a default value for `app_metadata`. Documentation does not say `app_metadata` can be `None`, as in `Optional[str]`.

I can make it None-safe, though.

Reply:
```suggestion
        if app_metadata is not None:
            self.endpoint.app_metadata = tobytes(app_metadata)
```

Reply:
Maybe we can write `str, default ""` instead? To me `str, optional` is a little ambiguous in this case: I would have taken it to mean that `None` is allowed. I don't see a clear spec for this in numpydoc but I did find this Pandas docstring guide (albeit from 2018) with a similar assumption

https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-3-parameters

> In cases where the default value is None, meaning that the value will not be used. Instead of “str, default None”, it is preferred to write “str, optional”.

Reply:
Works for me: 8c241c6593fa1fefdaafe4f63c4c41b10ff54cde